stl/src/filesystem.cpp

```diff
@@ -128,13 +128,14 @@ namespace {
 
     [[nodiscard]] __std_win_error __stdcall _Get_file_id_by_handle(
         const HANDLE _Handle, _Out_ FILE_ID_INFO* const _Id) noexcept {
-        __std_win_error _Last_error;
         if (GetFileInformationByHandleEx(_Handle, FileIdInfo, _Id, sizeof(*_Id)) != 0) {
             // if we could get FILE_ID_INFO, use that as the source of truth
             return __std_win_error::_Success;
         }
 
-        _Last_error = __std_win_error{GetLastError()};
+        __std_win_error _Last_error = __std_win_error{GetLastError()};

```

Comment:
As I mentioned before, we conventionally use direct-initialization here; `Type variable{initializer};` is simpler than `Type variable = Type{initializer};` with no loss of clarity. Precedent:

https://github.com/microsoft/STL/blob/9947dd9d6d2506029ed4486bb7596bda7d70f99d/stl/src/filesystem.cpp#L718